### DISCUSSION TITLE
Accessing the Assistant Context from a Terminal Script

### DISCUSSION THREAD

**Discussion by Cheirnomus** [zed-industries/zed#21226] [Feature Requests]
Working with Zed is a pleasure and I've designed a meta-prompt system that outputs multiple files that have different extension types depending on the objective, which can include building / stubbing out complete apps.

What I'd like to do next is run a terminal script that can extract the different tagged file type sections in the Assistant Panel's context, so that the script can then build a complete project from the Assistant's output.

The closest option I can find in the Zed documentation is using the Inline Assistant in the Terminal. This still does not sound like the solution I need, because it only pulls text from the Assistant's context to transform it via another prompt.

How would it be possible to run a Terminal script that can access the Assistant's context, so the script can then search the context and extract the tagged text sections it finds?

**Comment by Cheirnomus** [comment]
Can a Zed Task Variable be used to access the Assistant Editor context? 

The Tasks documentations says, "These variables allow you to pull information from the current editor and use it in your tasks".

Likely, this current editor does not include the Assistant Panel's context, right?

Maybe an Assistant History file can be accessed?

Can someone let me know what my best option is, thanks.

**Comment by anzax** [comment]
I was thinking along similar lines. I use Aider, and its approach to working with LLMs is very similar to Zed’s. The main focus for both is on managing context effectively. I believe it would be highly efficient to share the same context between Zed and Aider, allowing me to seamlessly switch between coding with Zed and delegating more autonomous tasks to Aider.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
